hazelcast-client-legacy/src/main/java/com/hazelcast/client/proxy/NearCachedClientMapProxy.java

```diff
@@ -235,24 +237,36 @@ protected void loadAllInternal(boolean replaceExistingValues, List<Data> dataKey
     }
 
     @Override
-    protected MapEntries getAllInternal(List<Data> keys, Map<K, V> result) {
-        Iterator<Data> iterator = keys.iterator();
-        while (iterator.hasNext()) {
-            Data key = iterator.next();
-            Object cached = nearCache.get(key);
-            if (cached != null && NULL_OBJECT != cached) {
-                result.put((K) toObject(key), (V) cached);
-                iterator.remove();
+    protected List<MapEntries> getAllInternal(Map<Integer, List<Data>> partitionToKeyData, Map<K, V> result) {
+        List<Integer> partitionsWithCachedEntries = new ArrayList<Integer>(partitionToKeyData.size());
+        for (Entry<Integer, List<Data>> partitionKeyEntry : partitionToKeyData.entrySet()) {
+            List<Data> keyList = partitionKeyEntry.getValue();
+            Iterator<Data> iterator = keyList.iterator();
+            while (iterator.hasNext()) {
+                Data key = iterator.next();
+                Object cached = nearCache.get(key);
+                if (cached != null && NULL_OBJECT != cached) {
+                    result.put((K) toObject(key), (V) cached);
+                    iterator.remove();
+                }
+            }
+            if (keyList.isEmpty()) {
```

Comment:
Do we really need to check for isEmpty condition? Lets say for partition 5 we have 10 entries and 9 of them are in cache, then all we want is to refresh the remaining 1 item, but in you implementation, it seems like we are trying to retrieve all 10 entries from the map for refreshing the cache. This is a change in behaviour. We can change it to refresh only the 1 item instead of the 9 items to keep the behaviour the same.